Stork service discovery + Openshift -  missing documentation related to required dependency on native mode

### Describe the bug

Related to: 
- https://github.com/quarkusio/quarkus-super-heroes/blob/43ded01f2a077a61507526bde07447d0276caaa0/rest-fights/pom.xml#L171-L185
 - https://github.com/quarkusio/quarkus/issues/25688
 - https://github.com/quarkusio/quarkus/issues/24444#issuecomment-1117136703

`io.smallrye.stork:stork-service-discovery-kubernetes` requires `io.quarkus:quarkus-openshift-client` or `io.quarkus:quarkus-kubernetes-client` in order to build a native image otherwise you will get the following native build error: 

```
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.jce.provider.BouncyCastleProvider
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 41 more
```

Another workaround is just to add the missing dependencies in your path 

```
<dependency>
            <groupId>org.bouncycastle</groupId>
            <artifactId>bcprov-jdk15on</artifactId>
        </dependency>
        <dependency>
            <groupId>org.bouncycastle</groupId>
            <artifactId>bcpkix-jdk15on</artifactId>
        </dependency>
```

I am missing all of these details on `io.smallrye.stork:stork-service-discovery-kubernetes`
Doc Ref:
- https://smallrye.io/smallrye-stork/1.1.2/service-discovery/kubernetes/
- https://smallrye.io/smallrye-stork/1.1.2/quarkus/

Maybe a simple note on Stork-quarkus would be enough


### How to Reproduce?

Just create a Quarkus application with `io.smallrye.stork:stork-service-discovery-kubernetes` and build a native image of your application -> `mvn clean package -Pnative`

### Output of `uname -a` or `ver`

Linux localhost.localdomain 5.18.7-100.fc35.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Jun 25 20:05:19 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 11.0.14 2022-01-18 OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9) OpenJDK 64-Bit Server VM Temurin-11.0.14+9 (build 11.0.14+9, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

### Additional information

_No response_

--- a/docs/src/main/asciidoc/stork-kubernetes.adoc
+++ b/docs/src/main/asciidoc/stork-kubernetes.adoc
@@ -429,6 +429,16 @@ The generated manifest can be applied to the cluster from the project root using
 kubectl apply -f target/kubernetes/kubernetes.yml -n=development
 ----
 
+
+[NOTE]
+====
+Please note that if you use Elliptic Curve keys with Stork and are getting exceptions like `java.lang.ClassNotFoundException: org.bouncycastle.jce.provider.BouncyCastleProvider`, then adding a BouncyCastle PKIX dependency (`org.bouncycastle:bcpkix-jdk15on`) is required.
+
+Note that internally an `org.bouncycastle.jce.provider.BouncyCastleProvider` provider will be registered if it has not already been registered.
+
+You can have this provider registered as described in the xref:security-customization.adoc#bouncy-castle[BouncyCastle] or xref:security-customization.adoc#bouncy-castle-fips[BouncyCastle FIPS] sections.
+====
+
 We're done!
 So, let's see if it works.
 

